From e4f8bf6058825819f87ebb477a1349cdbb930e66 Mon Sep 17 00:00:00 2001
From: persmule <persmule@hardenedlinux.org>
Date: Sun, 15 Dec 2024 21:03:57 +0800
Subject: [PATCH] hwdb: add scancodes for HP Elitebook 2170p runnning coreboot

HP Elitebook 2170p's keyboard scancodes under coreboot is different
with those under oem firmware. The scan code of backspace key of HP
Elitebook 2170p under coreboot is 0x66, but 93b078c has
KEYBOARD_KEY_66=pickup_phone for general HP laptops, making the
backspace key of the HP Elitebook 2170p running coreboot unusable.

The committed scancodes map KEYBOARD_KEY_66 back to backspace, and
provide all Fn-keys of HP Elitebook 2170p under coreboot.
Their evdev string matches the default SMBIOS tables of coreboot
for Elitebook 2170p, to avoid conflict with HP Elitebook 2170p running
oem firmware.

Fixes #35469 and https://ticket.coreboot.org/issues/571

(cherry picked from commit 93fc2adfa414e4a1b384da56c6e7d9277661ff4b)

Resolves: RHEL-49810
---
 hwdb.d/60-keyboard.hwdb | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index dba1274865..47d863f3b7 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -739,6 +739,19 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pnHP*G60*Notebook*PC:*
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pn*2570p*:*
  KEYBOARD_KEY_f8=wlan                                   # Wireless HW switch button
 
+# 2170p running coreboot
+evdev:atkbd:dmi:bvncoreboot:bvr*:bd*:svnHewlett-Packard*:pn*EliteBook2170p:*
+evdev:atkbd:dmi:bvncoreboot:bvr*:bd*:svnHP*:pn*EliteBook2170p:*
+ KEYBOARD_KEY_66=backspace
+ KEYBOARD_KEY_be=battery		# Fn+F2
+ KEYBOARD_KEY_d7=switchvideomode	# Fn+F4
+ KEYBOARD_KEY_a3=mute			# Fn+F5
+ KEYBOARD_KEY_a1=volumedown		# Fn+F6
+ KEYBOARD_KEY_a2=volumeup		# Fn+F7
+ KEYBOARD_KEY_a4=brightnessdown		# Fn+F9
+ KEYBOARD_KEY_c3=brightnessup		# Fn+F10
+ KEYBOARD_KEY_e3=wlan			# Fn+F12
+
 # Elitebook 2760p
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pn*2760p*:*
  KEYBOARD_KEY_89=battery                                # Fn+F8
